src/initialization.js

```diff
@@ -27,9 +29,30 @@ var initialization = {
                 isInitialized = true;
                 loadEventsAndPages();
             }
+
+            if (!window.optimizelyClientInstance) {
+                var instantiateFSClient = function() {
+                    var optimizelyClientInstance = window.optimizelySdk.createInstance({
+                        datafile: window.optimizelyDatafile
+                    });
+
+                    optimizelyClientInstance.onReady().then(() => {
+                        isInitialized = true;
+                        loadFullStackEvents();
+                    });  
+                }
+
+                helpers.loadScript('https://unpkg.com/@optimizely/optimizely-sdk/dist/optimizely.browser.umd.min.js', 
+                helpers.loadScript('https://cdn.optimizely.com/datafiles/' + settings.sdkKey + '.json/tag.js', instantiateFSClient));
+
+            } else {
+                isInitialized = true;
+                loadFullStackEvents();
+            }            
         } else {
             isInitialized = true;
             loadEventsAndPages();
+            loadFullStackEvents();
```

Comment:
i'd add a check here for each of these load events based on if they want to use optimizely web x and optimizely FS. we should only load them assuming certain fields are filled in

```suggestion
            isInitialized = true;
            if (settings.projectId) {
                loadEventsAndPages() // perhaps rename this function to be more specific to optimizely web X given we have `loadFullStackEvents` now also
            }
            if (settings.sdkKey) {
                loadFullStackEvents();
            }
```